controllers/argocd/util.go

```diff
@@ -1879,3 +1874,168 @@ func appendUniqueArgs(cmd []string, extraArgs []string) []string {
 
 	return cmd
 }
+
+func namespaceManagementFilterPredicate() predicate.Predicate {
```

Comment:
Need unit tests for this function, same comment as `argoCDNamespaceManagementFilterPredicate`